### PR TITLE
cpu/kinetis/include: fix xtimer backend timer selection

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

While testing #17357 I had the test fail on `frdm-kw41z`, while investigating I realized there was a wrong define and that the rtt callback was not cleared.

I also added a fix to the rtt_min_offset test which was not updated since now kinetis runs at 32khz.

### Testing procedure

`USEMODULE=evtimer_on_ztimer BOARD=frdm-kw41z make -C tests/evtimer_msg/ flash test -j` now passes

```
main(): This is RIOT! (Version: 2022.01-devel-1233-gbf0546-pr_kinetis_fix_rtt)
Testing generic evtimer
This should list 2 items
ev #1 offset=999
ev #2 offset=500
This should list 4 items
ev #1 offset=659
ev #2 offset=341
ev #3 offset=500
ev #4 offset=2454
Are the reception times of all 4 msgs close to the supposed values?
At   3854 ms received msg 0: "#2 supposed to be 3854"
.At   4195 ms received msg 1: "#0 supposed to be 4195"
.At   4695 ms received msg 2: "#1 supposed to be 4695"
.At   7149 ms received msg 3: "#3 supposed to be 7149"
.
All tests successful
```

- `BOARD=usb-kw41z make -C tests/periph_rtt_min/ flash term -j` default samples are used.

```
START
main(): This is RIOT! (Version: 2022.01-devel-1233-gbf0546-pr_kinetis_fix_rtt)
Evaluate RTT_MIN_OFFSET over 1024 samples
Sample 0
Sample 1
Sample 2
....
Sample 1021
Sample 1022
Sample 1023

RTT_MIN_OFFSET for usb-kw41z over 1024 samples: 1

```

### CODE REVIEW

File: cpu/kinetis/periph/rtt.c [hunk 1]
```diff
@@ -31,7 +31,7 @@
 #include "debug.h"
 
 static rtt_cb_t alarm_cb = NULL;       /**< callback called from RTC alarm */
-static void *alarm_arg;                /**< argument passed to the callback */
```
Comment (kaspar030) [inline cpu/kinetis/periph/rtt.c:34]: this shouldn't be necessary?

File: cpu/kinetis/periph/rtt.c [hunk 2]
```diff
@@ -31,7 +31,7 @@
...
+static void *alarm_arg = NULL;         /**< argument passed to the callback */
```
Comment (kaspar030) [inline cpu/kinetis/periph/rtt.c:34]: well, doesn't hurt either.

**Comment by kaspar030** [comment]
waiting for https://github.com/RIOT-OS/RIOT/pull/17394

**Comment by fjmolinas** [comment]
There is an unrelated failure... something wrong in master

**Comment by fjmolinas** [comment]
https://ci.riot-os.org/RIOT-OS/RIOT/17393/bf05468fab08691fed0abfc8da4a71075832c684/output/compile/tests/driver_lsm303dlhc/esp8266-esp-12x:gnu.txt

**Comment by fjmolinas** [comment]
All green @kaspar030 I split the waiting commit out

**Review by kaspar030 — APPROVED** [review]
ACK.

**Pull request merged**